flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/LogicalWindowJsonDeserializer.java

```diff
@@ -68,24 +66,25 @@ public LogicalWindowJsonDeserializer() {
     @Override
     public LogicalWindow deserialize(
             JsonParser jsonParser, DeserializationContext deserializationContext)
-            throws IOException, JsonProcessingException {
-        FlinkDeserializationContext flinkDeserializationContext =
-                (FlinkDeserializationContext) deserializationContext;
-        ObjectMapper mapper = flinkDeserializationContext.getObjectMapper();
+            throws IOException {
         JsonNode jsonNode = jsonParser.readValueAsTree();
         String kind = jsonNode.get(FIELD_NAME_KIND).asText().toUpperCase();
         WindowReference alias =
-                mapper.readValue(jsonNode.get(FIELD_NAME_ALIAS).toString(), WindowReference.class);
+                jsonParser
```

Comment:
What is the difference between this and
```
deserializationContext.readValue(
                                    jsonNode.get(FIELD_NAME_SIZE).traverse(jsonParser.getCodec()),
                                    Duration.class);
```
Why is this not consistent?

Reply:
Good question. For some reason, only in this specific case, I'm forced to use the codec directly. I feel like this has something to do with the fact that the class `WindowReference` is in scala. I also havent't spent much time investigating, as my understanding is that we don't need to serialize this class anymore once we disable `GroupWindow`s.